### PR TITLE
Add linked headers to write mode.

### PR DESCRIPTION
**Pull request by jmcarp**
[Resolves https://github.com/eregs/notice-and-comment/issues/203]
[Resolves https://github.com/eregs/notice-and-comment/issues/204]

screen shot 2016-04-26 at 1 43 48 pm

### CODE REVIEW

**Comment by cmc333333** [comment]
Seems like this logic should live in the backend rather than in JS, no?

**Comment by jmcarp** [comment]
@cmc333333 can you be more specific?

**Comment by cmc333333** [comment]
Sure! Sorry, wanted to finish before standup.

Rather than constructing the link and rendering it in Javascript, shouldn't it be part of the original template? I'd like to remove as much logic from JS as possible to make the app friendly for non-JS users.

File: regulations/static/regulations/js/source/views/comment/comment-view.js [hunk 1]
```diff
@@ -77,13 +82,32 @@ var CommentView = Backbone.View.extend({
...
+    var options = {
+      section: this.model.get('id'),
+      tocId: this.model.get('tocId'),
+      label: this.model.get('label')
+    };
+    // TODO: Push this logic into `PreambleView`
+    var type = options.section.split('-')[1];
+    DrawerEvents.trigger('section:open', options.tocId);
+    DrawerEvents.trigger('pane:change', type === 'preamble' ? 'table-of-contents' : 'table-of-contents-secondary');
+    MainEvents.trigger('section:open', options.section, options, 'preamble-section');
```
Comment (cmc333333) [inline regulations/static/regulations/js/source/views/comment/comment-view.js]: +1 to combining this with PreambleView; we basically want this to trigger identical events as if the user had clicked the "Read" tab.

**Comment by jmcarp** [comment]
At the moment, we're grabbing the relevant dom node from read mode, dropping markup related to commenting, and copying it into write mode. So if we're adding these links server-side, they'd show up in read mode as well, unless we expose a separate view to render a node with a header link (and without commenting markup). Is either of those (adding links to read mode, adding a new partial view) the direction you're thinking of here, or do we want to try something else?

**Comment by cmc333333** [comment]
Gotcha. I forgot how JS-centric the write mode was; another vote for eregs/notice-and-comment#164

Completely different: There's a duplication issue when commenting on a preamble header:
screen shot 2016-04-26 at 4 29 20 pm

I think the idea is to separate the "text" of the header from it's marker ("I.") If that's not easily accomplishable, maybe we drop the original header and make adding that text a separate issue?

**Comment by jmcarp** [comment]
At this point in the code, we have the human-readable label ("Section I") but not the title ("I. General information"). I'm assuming that we want the link to have the text "I. General information" in this case, but lmk if that's not right. Anyway, two hacks come to mind. First, like you said, we can check for a top-level header and drop it if we find one. Second, we can check for a top-level header and turn it into the link if it exists, else create a new link with the human-readable label as its text.

Preferences @cmc333333 ?

**Comment by cmc333333** [comment]
The downside to the second approach is that we wouldn't get "I.B.3", we'd only get "3". This may sound crazy, but what do you think about trying to align/overlap the two sources of data?
```
Section I.B.3
            3. Some Title
Section I.B.3. Some Title
```

**Comment by jmcarp** [comment]
This logic will probably need to get torn out and rewritten anyway, so why not? Pushed an implementation just now. It's clunky, as expected, and should go away as soon as possible.

**Pull request merged**